### PR TITLE
[ballot-selection] clean break from cvr selection

### PR DESCRIPTION
**Pull request by teaforthecat**
This is the big change that I was trying to avoid but I think there is light at the end of the tunnel.
There is an off by one error somewhere that is indicated by the smoke tests error messages. The smoke test passes though, so that is encouraging. It seemed good to get this in front of the team sooner rather than later.

### CODE REVIEW

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java [hunk 1]
```diff
@@ -527,96 +482,11 @@ public static boolean startNewRoundOfLength(final CountyDashboard the_cdb,
...
-                         generated_numbers,
-                         ballot_ids_to_audit,
-                         new_cvr_ids);
-      updateRound(the_cdb, the_cdb.currentRound());
-      updateCVRUnderAudit(the_cdb);
-      result = true;
-    }
-    return result;
+    final int expected_prefix_length = computeEstimatedSamplesToAudit(the_cdb);
+    return startNewRound(the_cdb, the_multiplier, expected_prefix_length);
```
Comment (teaforthecat) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java]: This is the interesting bit in all this: one function calls the other :o

File: server/eclipse-project/src/main/java/us/freeandfair/corla/query/BallotManifestInfoQueries.java [hunk 2]
```diff
@@ -182,6 +189,27 @@ Find the batch(bmi) that would hold the sequence number given.
...
+      final CriteriaBuilder cb = s.getCriteriaBuilder();
+      final CriteriaQuery<Long> cq = cb.createQuery(Long.class);
+      final Root<BallotManifestInfo> root = cq.from(BallotManifestInfo.class);
+      cq.select(cb.max(root.get("my_sequence_end")));
+      cq.where(cb.equal(root.get("my_county_id"), the_county));
+
+      final TypedQuery<Long> query = s.createQuery(cq);
+      result = OptionalLong.of(query.getSingleResult());
+    } catch (final PersistenceException e) {
+      // ignore
```
Comment (sduckett) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/query/BallotManifestInfoQueries.java]: I'd rather see some logging than a silent catch, even if it is `log.error("this is never going to happen");`

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java [hunk 3]
```diff
@@ -264,10 +268,10 @@ public String startSubsequentRound(final Request the_request, final Response the
         }
         if (start.useEstimates()) {
           round_started = 
-              ComparisonAuditController.startNewRoundFromEstimates(cdb, multiplier);
+              ComparisonAuditController.startNewRound(cdb, multiplier);
         } else {
           round_started = ComparisonAuditController.
-              startNewRoundOfLength(cdb, start.countyBallots().get(cdb.id()), multiplier);
+              startNewRound(cdb, multiplier, start.countyBallots().get(cdb.id()));
```
Comment (sduckett) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/StartAuditRound.java]: Thanks for aligning the arguments! 👏

File: server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRToAuditList.java [hunk 4]
```diff
@@ -157,32 +156,8 @@ public String endpointBody(final Request the_request, final Response the_respons
 
     try {
       // get the request parameters
-      final String start_param = the_request.queryParams(START);
```
Comment (sduckett) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRToAuditList.java]: Do all of these optional parameters only apply if we're selecting from the CVRs?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/CVRSelection.java [hunk 5]
```diff
@@ -1,62 +0,0 @@
-/**
```
Comment (sduckett) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/controller/CVRSelection.java]: 👋 ⛔️

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/BallotSelection.java [hunk 6]
```diff
@@ -81,17 +81,79 @@ private BallotSelection() {
...
     }
-    a_list.sort(new BallotOrderComparator());
     return a_list;
   }
 
+  /** turn round.ballotSequence() into a list of ballots for the user to audit **/
+  public static List<CVRToAuditResponse>
+      prepareBallotSequence(final List<Long> ballot_sequence) {
+    // this is what gets added to and returned
+    final List<CVRToAuditResponse> a_list = new LinkedList<CVRToAuditResponse>();
```
Comment (sduckett) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/controller/BallotSelection.java]: YMMV, but I'd call `a_list` something like `responses`.

**Comment by teaforthecat** [comment]
closing in favor of the smaller changes in #23

**Pull request closed without merging**